### ISSUE TITLE
Lingering "unclosed file" warnings

### ISSUE THREAD

**Issue by onyxfish** [wireservice/csvkit#693]
Was originally tracked as wireservice/agate#647, but really this was a csvkit issue. I've fixed most of them. Here are the remainder:

```
test_fixed_no_inference (tests.test_convert.test_fixed.TestFixed) ... /Users/cgroskopf/src/csvkit/csvkit/cli.py:124: ResourceWarning: unclosed file <_io.TextIOWrapper name='examples/testfixed_schema_no_inference.csv' mode='rt' encoding='utf-8'>
ok

test_invalid_options (tests.test_utilities.test_csvcut.TestCSVCut) ... /Users/cgroskopf/src/csvkit/csvkit/cli.py:294: ResourceWarning: unclosed file <_io.TextIOWrapper name='examples/dummy.csv' mode='rt' encoding='utf-8'>
ok

test_invalid_options (tests.test_utilities.test_csvgrep.TestCSVGrep) ... /Users/cgroskopf/src/csvkit/csvkit/cli.py:294: ResourceWarning: unclosed file <_io.TextIOWrapper name='examples/dummy.csv' mode='rt' encoding='utf-8'>
ok

test_duplicate_keys (tests.test_utilities.test_csvjson.TestCSVJSON) ... /Users/cgroskopf/src/csvkit/tests/test_utilities/test_csvjson.py:49: DeprecationWarning: Please use assertRaisesRegex instead.
ok

test_invalid_options (tests.test_utilities.test_csvstat.TestCSVStat) ... /Users/cgroskopf/src/csvkit/csvkit/cli.py:294: ResourceWarning: unclosed file <_io.TextIOWrapper name='examples/dummy.csv' mode='rt' encoding='utf-8'>
ok
```

**Comment by jpmckinney** [comment]
Can you PR your progress?

**Comment by onyxfish** [comment]
I already merged onto master. :|

**Comment by onyxfish** [comment]
All resolved except for the 3 relating to `test_invalid_options` which I'm having a hard time tracing. Seems like it has something to do with the exception handling, but I can't put my finger on it.

**Comment by jpmckinney** [comment]
I've fixed it in #695

**Comment by onyxfish** [comment]
Excellent! Merged in. Going to update changelog.
